src/resolve.cc

```diff
@@ -1,20 +1,50 @@
 #include "resolve.h"
 
 #include <iostream>
+#include <signal.h>
+#include <set>
+#include <vector>
+
+using std::set, std::string, std::cout, std::endl, std::vector, std::cerr;
+
+struct errsym {
+    enum errtype {
+        OK,
+        MULTIDEF,
+        NODEF
+    };
+    int err;    // Error type
+    string sym; // Error symbol name
+};
+
+errsym callResolveSymbols(std::vector<ObjectFile> &allObjects);
+
+void resolveSymbols(std::vector<ObjectFile> &allObjects) {
+    errsym ret = callResolveSymbols(allObjects);
+    switch (ret.err)
+    {
+    case errsym::MULTIDEF:
+        cerr << "multiple definition for symbol " << ret.sym << endl;
+        abort();
+        break;
+    case errsym::NODEF:
+        cerr << "undefined reference for symbol " << ret.sym << endl;
+        abort();
+        break;
+    default:
+        break;
+    }
```

Comment:
I don't like passing things around, packing symbol names in another structure and passing it out only for printing. Keep the skeleton code short and clean.